src/recommendationservice/recommendation_server.py

```diff
@@ -67,50 +76,59 @@ def Watch(self, request, context):
 def get_product_list(request_product_ids):
     global first_run
     global cached_ids
-    with tracer.start_as_current_span("get_product_list") as span:
-        max_responses = 5
-
-        # Formulate the list of characters to list of strings
-        request_product_ids_str = ''.join(request_product_ids)
-        request_product_ids = request_product_ids_str.split(',')
-
-        # Feature flag scenario - Cache Leak
-        if check_feature_flag("recommendationCache"):
-            span.set_attribute("app.recommendation.cache_enabled", True)
-            if random.random() < 0.5 or first_run:
-                first_run = False
-                span.set_attribute("app.cache_hit", False)
-                logger.info("get_product_list: cache miss")
-                cat_response = product_catalog_stub.ListProducts(demo_pb2.Empty())
-                response_ids = [x.id for x in cat_response.products]
-                cached_ids = cached_ids + response_ids
-                cached_ids = cached_ids + cached_ids[:len(cached_ids) // 4]
-                product_ids = cached_ids
-            else:
-                span.set_attribute("app.cache_hit", True)
-                logger.info("get_product_list: cache hit")
-                product_ids = cached_ids
-        else:
-            span.set_attribute("app.recommendation.cache_enabled", False)
+
+    # with tracer.start_as_current_span("get_product_list") as span:
+
+    max_responses = 5
+
+    # Formulate the list of characters to list of strings
+    request_product_ids_str = ''.join(request_product_ids)
+    request_product_ids = request_product_ids_str.split(',')
+
+    # Feature flag scenario - Cache Leak
+    if check_feature_flag("recommendationCache"):
+        # span.set_attribute("app.recommendation.cache_enabled", True)
+        newrelic.agent.add_custom_attribute("recommendation_cache_enabled", "True")
```

Comment:
To align this better with what the OTel instrumentation was doing it would be best to instrument `get_product_list` with the NR python agent. This way the shape of the telemetry emitted by the NR agent and otel would be the same. Furthermore, the `recommendation_cache_enabled` attribute would be applied to the correct span rather than the parent.